src/connection.ts

```diff
@@ -969,13 +970,41 @@ export class Connection<S extends Schema = Schema> extends EventEmitter {
       throw new Error('No SObject Type defined in record');
     }
     const url = [this._baseUrl(), 'sobjects', sobjectType].join('/');
+    let contentType, body;
+
+    if (options && options.multipartFileFields) {
+      // Send the record as a multipart/form-data request. Useful for fields containing large binary blobs.
+      const form = new FormData();
+      // Extract the fields requested to be sent separately from the JSON
+      Object.entries(options.multipartFileFields).forEach(
+        ([fieldName, fileDetails]) => {
+          form.append(
+            fieldName,
+            Buffer.from(rec[fieldName], 'base64'),
+            fileDetails,
+          );
+          delete rec[fieldName];
+        },
+      );
+      // Serialize the remaining fields as JSON
+      form.append(type, JSON.stringify(rec), {
```

Comment:
With my limited knowledge on TS... Logic is in this block of execution only when the conditional at #975 is satisfied, which establish that we are in this block to process multi-part requested fields.  So the the loop starting on #979 iterate to process each field that needs multi-part function.  If my assessment of this logic is correct, then i do not understand what is left to process on line #990?

Reply:
if there were fields that need JSON processing, shouldn't this branch into the "else" clause below?

Reply:
What remains is any fields that were not referenced in `multipartFileFields`. The key is the `delete rec[fieldName]` on line 986. Consider a hypothetical `rec` for an entity named `MyEntity" is passed in:
```
{
  Name: "MyRecordName",
  Cacheable: True,
  MyFileField: "<base64 encoded string or buffer>"
}
```

with `options.multipartFileFields`:
```
{
  MyFileFIeld: {
    contentType: "image/png",
    fileName: "company-logo.png"
  }
}
```

After the loop starting on line 979 completes, we have a `FormData` object that contains (when serialized):
```
----------------------------248819749774746952254207
Content-Disposition: form-data; name="MyFileField"; filename="company-logo.png" 
Content-Type: image/png
<base64 encoded file content>
```

We still need to add the remaining fields so line 990 serializes the next "part" of our request which results in:
```
----------------------------248819749774746952254207
Content-Disposition: form-data; name="MyEntity"
Content-Type: application/json
 
{"Name": "MyRecordName", "Cacheable": true}
```

Now all parts will be sent, with the appropriate content type.